Wip/rebased/dev

bewest opened the pull request:
This is an updated version of the dev branch made to be able to merge with master.

https://github.com/nightscout/minimed-connect-to-nightscout/pull/1

Apologies if the git trail is confusing, we had some diverging patches and branches at one point.  Our best efforts have been taken to preserve the history.

bewest committed:
Merge pull request #2 from FredMK/dev

confirmed working for user users as well

Commit:
Trying to solve ERR_TLS_CERT_ALTNAME_INVALID error

Commit:
Trying to solve ERR_TLS_CERT_ALTNAME_INVALID error2

Commit:
Trying to solve ERR_TLS_CERT_ALTNAME_INVALID error3

Commit:
test

Commit:
process.env["NODE_TLS_REJECT_UNAUTHORIZED"] = 0;

Commit:
test

Commit:
test2

Commit:
test

Commit:
checkServerIdentity

Commit:
why?

Commit:
working

bewest committed:
allow specifying server name via environment variable

Commit:
refactor for sessonId

bewest committed:
pass runtime country and lang values

Future-proofing a bit.

bewest committed:
be consistent about using qs and url from stdlib

Commit:
refactor with axios

bewest committed:
1.4.0

FredMK committed:
Merge pull request #2 from nightscout/master

From master

FredMK committed:
connection test for EU and US server

FredMK committed:
connection test for EU and US server

FredMK committed:
connection test for EU and US server

FredMK committed:
servername env support

FredMK committed:
missing function fix

FredMK committed:
missing function fix

FredMK committed:
missing function fix

FredMK committed:
callback enh

FredMK committed:
callback enh

FredMK committed:
callback enh

saukk900 committed:
Add new node version to engines

Commit:
error fix

FredMK committed:
axious timout

FredMK committed:
axious timout

FredMK committed:
axious timout

FredMK committed:
Logging enh

FredMK committed:
Logging enh

FredMK committed:
Logging enh

FredMK committed:
User agent not axios (refresh token not working with default)
minor changes

FredMK committed:
User agent not axios (refresh token not working with default)
minor changes

FredMK committed:
User agent not axios (refresh token not working with default)
minor changes

FredMK committed:
dynamic interval

FredMK committed:
dynamic interval

FredMK committed:
dynamic interval

FredMK committed:
Heroku debug

FredMK committed:
Heroku debug

FredMK committed:
Heroku debug

FredMK committed:
heroku debug 2

FredMK committed:
heroku debug 2

FredMK committed:
heroku debug 2

FredMK committed:
revert debug

FredMK committed:
revert debug

FredMK committed:
revert debug

FredMK committed:
fresh token error fix

FredMK committed:
fresh token error fix

FredMK committed:
fresh token error fix

FredMK committed:
Request count safety check

FredMK committed:
Request count safety check

FredMK committed:
Request count safety check

FredMK committed:
interval fix on stale data

FredMK committed:
interval fix on stale data

FredMK committed:
interval fix on stale data

bewest committed:
typo in US test name, allow tests time to pass

testing against international servers takes longer.

bewest committed:
typo in US test name, allow tests time to pass

testing against international servers takes longer.

bewest committed:
Merge pull request #10 from nightscout/wip/stability/axios

@fredmk's axios refactor rebased on dev

bewest committed:
apparently not needed anymore

bewest committed:
apparently not needed anymore

bewest committed:
Merge pull request #16 from nightscout/wip/stability/axios

apparently not needed anymore

bewest committed:
Merge remote-tracking branch 'origin/dev' into rebased_dev

bewest committed:
Merge branch 'add_node_version' of https://github.com/saukk900/minimed-connect-to-nightscout into saukk900/wip/add_node_version

bewest committed:
include software name, version, and contact

This is intended to provide better traceability to this software.

mrinnetmaki committed:
Use an axios instance per client

Add an axios instance for the client, rather than changing global settings of axios.
This allows an app to create multiple clients (for different nightscout users) and to use them in parallel.

mrinnetmaki commented on carelink.js:
```diff
@@ -24,21 +33,27 @@ var CARELINK_SECURITY_URL = 'https://' + carelinkServerAddress + '/patient/j_sec
     }
 
     axiosCookieJarSupport(axios);
-    var cookieJar = new tough.CookieJar();
```
Agree the cookie jar is now initiated two times on consecutive rows. However, if you remove the var `cookieJar` here, it is undefined in row 281...

bewest replied:
I believe this was due to a mistake on my end while resolving conflicts and should be fixed.

mrinnetmaki reviewed (commented):
Good stuff @bewest!

I had another (or perhaps supplementary) approach for a safe multi-tenant implementation (regarding your comment https://github.com/nightscout/minimed-connect-to-nightscout/issues/11#issuecomment-739419508).

I believe we should use an axios instance per client. See #18.

Just pointing that out in case it would trigger any thoughts on the logins and cookie handling.

bewest committed:
store temp merge results

bewest committed:
matches @fredmk's axios branch much closer

bewest committed:
Merge branch 'wip/rebased/dev' into wip/rebased/axios-instance-per-client

bewest commented:
Thanks, @mrinnetmaki I was able to fix that error as well as rebase your idea over here: https://github.com/nightscout/minimed-connect-to-nightscout/pull/19

Something is not quite right for the US test though, even though it is fine in this branch.

FredMK committed:
console.log on refresh EU token fail
comment fix

FredMK committed:
Merge branch 'wip/rebased/dev' into axios2

# Conflicts:
#	carelink.js
#	test/connectUs.js

FredMK committed:
Merge remote-tracking branch 'upstream/wip/rebased/axios-instance-per-client' into axios2

bewest committed:
force oauth method everywhere

In order to use oauth everywhere correctly, the countrycode must be correctly
set AND the correct sovereign server must be used.

bewest committed:
Merge branch 'wip/oauth/everywhere' into wip/rebased/axios-instance-per-client

FredMK committed:
User-Agent set just before request is sent (avoid code duplication)

FredMK committed:
minor refactor

FredMK committed:
better environment variable parsing

FredMK committed:
Random timeout on start to avoid ban for managed environments

bewest committed:
make multitenant safe, use oauth everywhere

bewest committed:
Merge pull request #19 from nightscout/wip/rebased/axios-instance-per-client

Wip/rebased/axios instance per client

bewest commented:
I don't know if this works in Argentina, but this seems to work in the US and in EU.  Thanks for the extra patches, @mrinnetmaki.

bewest committed:
Merge pull request #20 from FredMK/axios2

Axios2

bewest committed:
set empty cookie on token refresh

Per https://github.com/nightscout/minimed-connect-to-nightscout/issues/15.
Recent feedback indicates that people trying without this are receiving 403.   Per the issue details, setting cookie at this point may allow things to work.
Unit tests pass either way.
